File: Decorator/typescript/src/Condiments/Whip.ts
import { Beverage } from "../Beverages";
import CondimentDecorator from "./CondimentDecorator";

export class Whip extends CondimentDecorator {
  private _beverage: Beverage;

  constructor(beverage: Beverage) {
    super();
    this._beverage = beverage;
    this.description = `${this._beverage.description}, Whip`;
  }

  cost() {
    return 0.1 + this._beverage.cost();
  }
}

File: Decorator/c#/Condiments/Whip.cs
using Beverages;

namespace Condiments
{
  public class Whip : CondimentDecorator
  {
    private Beverage _beverage;

    public Whip(Beverage beverage)
    {
      _beverage = beverage;
      description = $"{_beverage.description}, Whip";
    }

    public override decimal cost()
    {
      return .10m + _beverage.cost();
    }
  }
}